integrations/visual_studio/creating_packages.rst

```diff
@@ -0,0 +1,228 @@
+
+::
+
+    build\HelloLib\HelloLib.sln
+
+You should be able to select the ``Greet`` subproject -> ``Set as Startup Project``.
+Then build and run the app with Ctrl+F5. (Debug -> Start Without Debugging)
+
+
+.. code-block:: bash
+
+    $ Hello World Debug!
+    # Switch IDE to Release mode, repeat
+    $ Hello World Release!
+
+Because the ``hello.cpp`` file contains an ``#ifdef _DEBUG`` to switch between debug and release
+message.
+
+In the repository, there is already a ``conanfile.py`` recipe:
+
+.. code-block:: python
+
+    from conans import ConanFile
+    from conan.tools.microsoft import MSBuild
+
+
+    class HelloConan(ConanFile):
+        name = "Hello"
+        version = "0.1"
+        license = "MIT"
+        url = "https://github.com/conan-io/examples"
+        settings = "os", "compiler", "build_type", "arch"
+        exports_sources = "src/*", "build/*.vcxproj*", "build/*.sln*"
+
+        def build(self):
+            msbuild = MSBuild(self)
+            msbuild.build("build/HelloLib/HelloLib.sln")
+
+        def package(self):
+            self.copy("*.h", dst="include", src="src")
+            self.copy("*.lib", dst="lib", keep_path=False)
+            self.copy("*.dll", dst="bin", keep_path=False)
+
+        def package_info(self):
+            self.cpp_info.libs = ["HelloLib"]
+
+
+This recipe is using the (:ref:)`MSBuild() build helper <msbuild>` to build the ``sln`` project.
+If our recipe has ``requires``, the ``MSBUILD`` helper will also take care of inject all the needed
+information from the requirements, as include directories, library names, definitions, flags etc
+to allow our project to locate the declared dependencies.
+
+The recipe contains also a ``test_package`` folder with a simple example consuming application.
+In this example, the consuming application is using CMake to build, but it could also use Visual
+Studio too. We have left the CMake one because it is the default generated with
+:command:`conan new`, and also to show that packages created from Visual Studio projects can also
+be consumed with other build systems like CMake.
+
+Once we want to create a package, it is advised to close VS IDE, clean the temporary build files
+from VS to avoid problems, then create and test the package. Here it is using system defaults,
+assuming they are Visual Studio 19, Release, x86_64:
+
+.. code-block:: bash
+
+   # close VS
+   $ git clean -xdf
+   $ conan create . demo/testing
+   ...
+   > Hello World Release!
+
+Instead of closing the IDE and running the command:`git clean` we could also configure a smarter
+filter in ``exports_sources`` field, so temporary build files are not exported into the recipe.
+
+This process can be repeated to create and test packages for different configurations:
+
+.. code-block:: bash
+
+   $ conan create . demo/testing -s arch=x86
+   $ conan create . demo/testing -s compiler="Visual Studio" -s compiler.runtime=MDd -s build_type=Debug
```

Comment:
Lets move all examples to the new ``msvc`` compiler setting, it is extremely likely that ``Visual Studio`` will be removed in 2.0

Reply:
Done